File: api-openbis-java/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/pat/search/PersonalAccessTokenSearchCriteria.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.pat.search;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.AbstractObjectSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchCriteriaToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchOperator;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.pat.id.IPersonalAccessTokenId;
import ch.systemsx.cisd.base.annotation.JsonObject;

@JsonObject("as.dto.pat.search.PersonalAccessTokenSearchCriteria")
public class PersonalAccessTokenSearchCriteria extends AbstractObjectSearchCriteria<IPersonalAccessTokenId>
{

    private static final long serialVersionUID = 1L;

    public PersonalAccessTokenSearchCriteria()
    {
    }

    public PersonalAccessTokenSearchCriteria withOrOperator()
    {
        return (PersonalAccessTokenSearchCriteria) withOperator(SearchOperator.OR);
    }

    public PersonalAccessTokenSearchCriteria withAndOperator()
    {
        return (PersonalAccessTokenSearchCriteria) withOperator(SearchOperator.AND);
    }

    public PersonalAccessTokenOwnerSearchCriteria withOwner()
    {
        return with(new PersonalAccessTokenOwnerSearchCriteria());
    }

    public PersonalAccessTokenSessionNameSearchCriteria withSessionName()
    {
        return with(new PersonalAccessTokenSessionNameSearchCriteria());
    }

    @Override
    protected SearchCriteriaToStringBuilder createBuilder()
    {
        SearchCriteriaToStringBuilder builder = super.createBuilder();
        builder.setName("PERSONAL_ACCESS_TOKEN");
        return builder;
    }

}

File: api-openbis-javascript/src/v3/as/dto/pat/search/PersonalAccessTokenSearchCriteria.js
/**
 * @author pkupczyk
 */
define([ "require", "stjs", "as/dto/common/search/AbstractObjectSearchCriteria", "as/dto/common/search/AbstractCompositeSearchCriteria", "as/dto/pat/search/PersonalAccessTokenOwnerSearchCriteria", "as/dto/pat/search/PersonalAccessTokenSessionNameSearchCriteria" ], function(require, stjs, AbstractObjectSearchCriteria) {
	var PersonalAccessTokenSearchCriteria = function() {
		AbstractObjectSearchCriteria.call(this);
	};
	stjs.extend(PersonalAccessTokenSearchCriteria, AbstractObjectSearchCriteria, [ AbstractObjectSearchCriteria ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.pat.search.PersonalAccessTokenSearchCriteria';
		constructor.serialVersionUID = 1;
		prototype.withOwner = function() {
			var PersonalAccessTokenOwnerSearchCriteria = require("as/dto/pat/search/PersonalAccessTokenOwnerSearchCriteria");
			return this.addCriteria(new PersonalAccessTokenOwnerSearchCriteria());
		};
		prototype.withSessionName = function() {
			var PersonalAccessTokenSessionNameSearchCriteria = require("as/dto/pat/search/PersonalAccessTokenSessionNameSearchCriteria");
			return this.addCriteria(new PersonalAccessTokenSessionNameSearchCriteria());
		};
	}, {
		criteria : {
			name : "Collection",
			arguments : [ "ISearchCriteria" ]
		}
	});
	return PersonalAccessTokenSearchCriteria;
})